### ISSUE TITLE
Create a DevTools extension for `package:provider`

### ISSUE THREAD

**Issue by kenzieschmoll** [rrousselGit/provider#833]
This FR tracks the work to migrate the DevTools Provider screen over to the DevTools Extensions framework.

The tracking issue for DevTools extension support is here: https://github.com/flutter/devtools/issues/1632. 
The tracking issue to remove the existing provider screen from DevTools is here: https://github.com/flutter/devtools/issues/6215
A draft PR to create the DevTools extension in package:provider is here: https://github.com/rrousselGit/provider/pull/832

It may make sense to restructure the repo to look like
```
provider/
|-packages/
    |-provider/  # existing provider code
    |  |-extension/
    |     |-devtools/
    |        |-build/   # build output of provider_devtools_extension flutter web app
    |        |-config.json
    |-provider_devtools_extension/   # the devtools extension web app
```
this would ensure the extension app is not included with the provider pub package, but rather only the build output needed for loading the devtools extension in devtools. This would also create a more scalable solution if, for example, there was a need for more sibling packages, like a provider analyzer plugin. @rrousselGit I will leave the design decision on that up to you, but that is my recommendation.

**Comment by rrousselGit** [comment]
Nice! I'll happily look into it

**Comment by kenzieschmoll** [comment]
An early version of `package:devtools_extensions` and `package:devtools_app_shared` are now published to pub, which means you should be able to pull https://github.com/rrousselGit/provider/pull/832 down to play with without having to do some hacky configuration. That PR also includes a launch config for VS code that allows you to launch the extension in a simulated environment without having to do a full DevTools build. 

The latest instructions for extension development are here: https://pub.dev/packages/devtools_extensions

**Comment by kenzieschmoll** [comment]
This is complete and published on version [6.1.0-dev.1](https://pub.dev/packages/provider/versions/6.1.0-dev.1)
